src/Config/RuntimeConfigLoader.cs

```diff
@@ -96,6 +105,65 @@ public static bool TryParseConfig(string json, [NotNullWhen(true)] out RuntimeCo
         return true;
     }
 
+    /// <summary>
+    /// Parses a JSON string into a <c>RuntimeConfig</c> object for multiple database scenario.
+    /// </summary>
+    /// <param name="json">JSON that represents the config file.Json expected to contain Datasources dictionary.</param>
+    /// <param name="config">The parsed config, or null if it parsed unsuccessfully.</param>
+    /// <returns>True if the config was parsed, otherwise false.</returns>
+    public static bool TryParseConfigMultipleDatabase(string json, [NotNullWhen(true)] out RuntimeConfig? config, ILogger? logger = null, Dictionary<string, string>? datasourceNameToConnectionString = null)
+    {
+        JsonSerializerOptions options = GetSerializationOptions();
+
+        try
+        {
+            config = JsonSerializer.Deserialize<RuntimeConfig>(json, options);
+
+            if (config is null)
+            {
+                return false;
+            }
+
+            if (datasourceNameToConnectionString != null)
+            {
+                foreach ((string dataSourceName, string connectionString) in datasourceNameToConnectionString)
+                {
+                    DataSource ds = config.DatasourceNameToDataSource[dataSourceName];
```

Comment:
Are we certain that the keys from the input dictionary will be the same in the DatasourceNameToDataSource member?

Reply:
Yes they are supposed to be. Ive added a conditon to validate this. DataSourceNameToDatasource would be a created dictionary part of the config that is tightly couples to the entityNametodatasource dict. Hence whenever the connection strings are added they will have to be added to the correct datasources in the datasourcenametodatasource dict only.